docs/local-ui-dev.md

```diff
@@ -0,0 +1,85 @@
+# Local UI Development
+
+This directory contains tools and instructions for using the RE api in a local development context which allows close operation with other kbase services in a custom docker network.
+
+This is enabled by running on the docker network `kbase-dev`.
+
+The `docker-compose.yaml` in this directory takes care of that.
+
+In addition, the configuration ensures that the entire schema set is loaded into Arango. This facilitates the usage of any collection during development.
+
+You'll need to take care of data loading.
+
+## Build and start the image
+
+From the repo root:
+
+```bash
+make start-dev
+```
+
+## Install your data
+
+At present, only instructions for installing the `data_sources` are provided. Clearly a complete local workflow would involve installing major chunks of the RE graph.
+
+see [importers/data_sources/README.md](../importers/data_sources/README.md#procedure).
+
+## Point services to relation-engine-api
+
+Within the `kbase-dev` docker network, the RE api is associated with the host `relation-engine-api`, and is bound to port `5000` as KBase services typically are.
+
+### Example: Taxonomy RE API
```

Comment:
Are there complementary docs in the taxonomy and kbase-ui repos? If not, I think it would be far more useful to have the docs there as people are more likely to be coming to the RE repo from those projects, rather than the other way around.

Reply:
I find it more convenient to have instructions for different development scenarios in the service repo itself, but ymmv. 

It seems to me the service repo should have documentation describing different development scenarios. As a kbase-ui developer, I'm already aware that generally I can work ui and service components in an integrated manner - how easy that is depends on whether the service already provides that support (and that is not universal by any means.)
But it is a good point that the kbase-ui docs could have instructions for developing with specific services.

Reply:
There are arguments for both sides -- I think the best solution is to make sure that docs are available in both repos. You're already very experienced with kbase-ui, so you will know all the ins and outs of it; other users (e.g. myself) might have set up kbase-ui locally but not know about additional services that can be enabled.

Reply:
Consider from the other point of view.
There are not too many development scenarios for an individual service, and having them documented in the service can be very useful. I can't count the number of times I've consulted the documentation in the narrative related to local development in docker which I added many years ago, without complaint.
There are many services a user interface may interface with, and many of them have their own idiosyncratic tools, scripts, docker builds, dependencies.

Reply:
@eapearson have you added docs to the appropriate repos? Can you point me to them?